### DISCUSSION TITLE
ESP-32 enable BLE iBeacon

### DISCUSSION THREAD

**Discussion by marcolino7** [arendst/Tasmota#13969] [Q&A] [answered]
Hi,
I would like to use some of my ESP32 node around my house, to detect presence in a room.
I would like to use BLE detecting a iBeacon token. I would like to use my mobile phone as iBeacon Token, using this app: https://play.google.com/store/apps/details?id=net.alea.beaconsimulator&hl=it&gl=US

To achieve this, I dowloaded Tasmota source, enabled `#define USE_IBEACON_ESP32 `, and compiled with PlatformIO using task tasmota-32 bluetooth and flashed.
After the I enabled BLE with command `SetOption115 1`.
Then I created an iBeacon token on my phone and enabled it, but in tele/nodename/SENSOR i do not get IBEACON information about my token.
Looking at the tele messages on MQTT I only get this:
`tele/tasmota_8B3820/BLE
{
  "Time": "2021-12-08T22:07:14",
  "BLE": {
    "scans": 696,
    "adverts": 326500,
    "devices": 4,
    "resets": 0
  }
}
`
and in console this:
`22:07:14.836 MQT: tele/tasmota_8B3820/SENSOR = {"Time":"2021-12-08T22:07:14","ESP32":{"Temperature":53.3},"Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":0},"TempUnit":"C"}
22:07:14.868 MQT: tele/tasmota_8B3820/BLE = {"Time":"2021-12-08T22:07:14","BLEDevices":{"total":4,"30F772499xxx":{"i":0,"r":-98},"B8BC5BB4Dxxx":{"i":1,"r":-85},"600308C1Axxx":{"i":2,"r":-82},"880F10921xxx":{"i":3,"r":-96}}}
22:07:14.892 MQT: tele/tasmota_8B3820/BLE = {"Time":"2021-12-08T22:07:14","BLE":{"scans":696,"adverts":326500,"devices":4,"resets":0}}
22:07:22.791 EQ3 poll cycle starting`

Any help would be appreciated.

Marco

**Comment by marcolino7** [comment]
Hi,
I did one step ahead. In console I do command:
`IBEACON 1`

and now i see ESP scanning for IBeacon, and I receive one related to a Sony TV, perhaps of someone live near my house, but I cannot receive the one generated by my phone.
Anyone have experience about this?

Thanks

**Comment by ascillato2** [comment, marked as answer]
Since you haven't had here any answer yet, have you tried in the Discord [Chat](https://discord.gg/Ks2Kzd4)?

**Comment by ingrove** [comment]
I followed the same steps as marcolino7 and achieved the same results.  I tried 2 different iBeacon generators, the one used by marcolino7 and the iBeacon transmitter embedded in the Home Assistant Android Companion App.  Neither were detected.  I flashed ESPresense onto the ESP32's and they detected both iBeacons immediately.  I'm using a Google Pixel 5 running Android 12.  Would really love to use Tasmota instead of ESPresense.
